framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -196,3 +196,40 @@ MaterialPropertyInterface::checkExecutionStage()
     mooseError("Material properties must be retrieved during object construction to ensure correct "
                "problem integrity validation.");
 }
+
+const MaterialProperty<Real> &
+MaterialPropertyInterface::getMaterialPropertyDot(const std::string & name)
+{
+  if (!_stateful_allowed)
+    mooseError("Stateful material properties not allowed for this object."
+               " Dot property for \"",
+               name,
+               "\" was requested.");
+
+  // Check if the supplied parameter is a valid input parameter key
+  std::string prop_name = deducePropertyName(name);
+
+  // Check if it's just a constant
+  const MaterialProperty<Real> * default_property = defaultMaterialProperty<Real>(prop_name);
+  if (default_property)
+    return *default_property;
```

Comment:
If it's constant then you should return `0`.

Reply:
oh, right!